integrations/amazon_sagemaker/tests/test_sagemaker.py

```diff
@@ -1,243 +1,175 @@
 import os
-from unittest.mock import Mock
+from unittest.mock import Mock, patch
 
 import pytest
+from botocore.exceptions import BotoCoreError
+from haystack.utils.auth import EnvVarSecret
 from haystack_integrations.components.generators.amazon_sagemaker import SagemakerGenerator
 from haystack_integrations.components.generators.amazon_sagemaker.errors import AWSConfigurationError
 
+mocked_dict = {
+    "type": "haystack_integrations.components.generators.amazon_sagemaker.sagemaker.SagemakerGenerator",
+    "init_parameters": {
+        "model": "model",
+        "aws_access_key_id": {"type": "env_var", "env_vars": ["AWS_ACCESS_KEY_ID"], "strict": False},
+        "aws_secret_access_key": {"type": "env_var", "env_vars": ["AWS_SECRET_ACCESS_KEY"], "strict": False},
+        "aws_session_token": {"type": "env_var", "env_vars": ["AWS_SESSION_TOKEN"], "strict": False},
+        "aws_region_name": {"type": "env_var", "env_vars": ["AWS_DEFAULT_REGION"], "strict": False},
+        "aws_profile_name": {"type": "env_var", "env_vars": ["AWS_PROFILE"], "strict": False},
+        "aws_custom_attributes": {"accept_eula": True},
+        "generation_kwargs": {"max_new_tokens": 10},
+    },
+}
```

Comment:
I would put this directly in the tests that use it, even if means duplicating it. 
Makes it easier to maintain and reason about.

As an example see this 👉  https://github.com/deepset-ai/haystack-core-integrations/blob/d28ffa1b7ab4a6e8363da2644996b70150145faa/integrations/google_ai/tests/generators/test_gemini.py#L97-L117